packages/astro/src/runtime/server/render/csp.ts

```diff
@@ -35,8 +35,14 @@ export function renderCspContent(result: SSRResult): string {
 		styleResources = result.styleResources.map((r) => `${r}`).join(' ');
 	}
 
+	let fontResources = "'self'";
+	if (result.fontResources.length > 0) {
+		fontResources = result.fontResources.map((r) => `${r}`).join(' ');
+	}
```

Comment:
Font resources are a bit different from the others. If the user doesn't use Astro Fonts **and** and they don't use `csp.fontResources`, `result.fontResources` is empty; however, we will end up with a `font-src: 'self'` even though no fonts were used.

This means that we should add `font-src` only when `result.fontResources` contains entries. Would you mind adding a new test (or updating an existing one), and assert that *when we don't use fonts, but we use CSP, the `font-src` directive isn't present*.